hazelcast/src/main/java/com/hazelcast/internal/server/tcp/LocalAddressRegistry.java

```diff
@@ -128,7 +128,23 @@ public void register(@Nonnull UUID instanceUuid, @Nonnull LinkedAddresses linked
                             + " member.");
                 }
             }
-            linkedAddresses.getAllAddresses().forEach(address -> addressToUuid.put(address, instanceUuid));
+            linkedAddresses.getAllAddresses().forEach(address -> {
+                if (addressToUuid.containsKey(address)) {
```

Comment:
Would it be possible to check these corner cases with tests too? I don't mean checking the log message, but just having a test case where the source and target clusters have an overlapping address.

Reply:
In this scenario, the members start to send operations to wrong members which is really problematic. I don't really know how to improve this case (and I don't know what to check when creating this kind of scenario). We can avoid overriding the addresses in this case, but, regardless it can cause separate problems when members forcefully restart.